### PR TITLE
update documentation for 3x API

### PR DESCRIPTION
**Pull request by chensuyue**
## Type of Change

documentation 

## Description

- [x] Main page docs matrix, @yiliu30 please update description for `docs/3x/design.md`. 
- [ ] ~~Main page get started examples, @yiliu30~~ Replace auto-round example with fp8 @xin3he in another PR.
- [x] 2.x feature overview md, `2x_user_guide.md`.
- [ ] Installation Guide update for 3x API, will be submitted in other PR.
- [x] docs for IO page, `get_started.md`.

### CODE REVIEW

File: docs/3x/get_started.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+# Getting Started
+
+1. [Quick Samples](#quick-samples)
+
+2. [Feature Matrix](#feature-matrix)
+
+## Quick Samples
```
Comment (yiliu30) [inline docs/3x/get_started.md:7]: We'd better have a quickstart for tf.

File: docs/3x/get_started.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+```shell
+# Install Intel Neural Compressor
```
Comment (yiliu30) [inline docs/3x/get_started.md:10]: Please add this example to the regular test.

File: docs/3x/get_started.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+from neural_compressor.torch.quantization import RTNConfig, prepare, convert
+
+user_model = AutoModelForCausalLM.from_pretrained("EleutherAI/gpt-neo-125m")
+quant_config = RTNConfig()
+prepared_model = prepare(model=user_model, quant_config=quant_config)
+quantized_model = convert(model=prepared_model)
+```
+
+## Feature Matrix
+Intel Neural Compressor 3.X aim to provide framework extension APIs for the popular deep learning frameworks PyTorch and TensorFlow.
```
Comment (yiliu30) [inline docs/3x/get_started.md:24]:
```suggestion
Intel Neural Compressor 3.X aims to extend PyTorch and TensorFlow's APIs to support compression techniques.
```

File: docs/3x/get_started.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+user_model = AutoModelForCausalLM.from_pretrained("EleutherAI/gpt-neo-125m")
+quant_config = RTNConfig()
+prepared_model = prepare(model=user_model, quant_config=quant_config)
+quantized_model = convert(model=prepared_model)
+```
+
+## Feature Matrix
+Intel Neural Compressor 3.X aim to provide framework extension APIs for the popular deep learning frameworks PyTorch and TensorFlow. 
+The below table provides a quick overview of the APIs available in Intel Neural Compressor 3.X.
+The new APIs mainly focus on quantization related features especially for the algorithms benefit LLM inference.
```
Comment (yiliu30) [inline docs/3x/get_started.md:26]:
```suggestion
The Intel Neural Compressor 3.X mainly focuses on quantization-related features, especially for algorithms that benefit LLM accuracy and inference.
```

File: docs/3x/get_started.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+quant_config = RTNConfig()
+prepared_model = prepare(model=user_model, quant_config=quant_config)
+quantized_model = convert(model=prepared_model)
+```
+
+## Feature Matrix
+Intel Neural Compressor 3.X aim to provide framework extension APIs for the popular deep learning frameworks PyTorch and TensorFlow. 
+The below table provides a quick overview of the APIs available in Intel Neural Compressor 3.X.
+The new APIs mainly focus on quantization related features especially for the algorithms benefit LLM inference. 
+It also provides some common APIs across different frameworks. For example, Auto Tune API support accuracy driven quantization and mixed precision, benchmark API aimed to measure the multiple instances performance of the quantized model.
```
Comment (yiliu30) [inline docs/3x/get_started.md:27]: I feel `module` is more appropriate than `APIs`. BTW, `common` is invisible for users.

**Review by yiliu30 — CHANGES_REQUESTED** [review]

File: docs/3x/get_started.md (hunk 2 above)
Comment (chensuyue) [reply docs/3x/get_started.md:10]: Ok, I will.

File: docs/3x/get_started.md (hunk 1 above)
Comment (chensuyue) [reply docs/3x/get_started.md:7]: I think we don't need much examples in get started, a simple one should be ok.

File: docs/3x/get_started.md (hunk 5 above)
Comment (chensuyue) [reply docs/3x/get_started.md:27]: Agree, I will change APIs into module. And here the common APIs not related to the common folder, just means auto-tune and benchmark which is common used across frameworks. To avoid confuse, I will update into others.

**Review by yiliu30 — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -116,54 +116,58 @@ quantized_model = fit(model=float_model, conf=static_quant_conf, calib_dataloade
...
+          <td colspan="4" align="center"><a href="./docs/3x/autotune.md">Auto Tune</a></td>
+          <td colspan="4" align="center"><a href="./docs/3x/benchmark.md">Benchmark</a></td>
       </tr>
   </tbody>
 </table>
 
-> **Note**: 
-> Further documentations can be found at [User Guide](https://github.com/intel/neural-compressor/blob/master/docs/source/user_guide.md).
+> **Note**:   
+> From 3.0 release, we recommend to use 3.X API. Compression techniques during training such as QAT, Pruning, Distillation only available in [2.X API](https://github.com/intel/neural-compressor/blob/master/docs/source/2x_user_guide.md).
```
Comment (thuang6) [inline README.md:170]: "Compression techniques during training such as QAT, Pruning, Distillation only available in [2.X API]" , suggest add "currently"

File: docs/3x/get_started.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+  <tbody>
+      <tr>
+          <td colspan="4" align="center"><a href="autotune.md">Auto Tune</a></td>
+          <td colspan="4" align="center"><a href="benchmark.md">Benchmark</a></td>
+      </tr>
+  </tbody>
+</table>
+
+> **Note**:   
+> From 3.0 release, we recommend to use 3.X API. Compression techniques during training such as QAT, Pruning, Distillation only available in [2.X API](https://github.com/intel/neural-compressor/blob/master/docs/source/2x_user_guide.md).
```
Comment (thuang6) [inline docs/3x/get_started.md:88]: same, suggest add "currently"

**Review by thuang6 — APPROVED** [review]
